File: src/main/kotlin/magicbook/gtlitecore/integration/IntegrationModule.java
package magicbook.gtlitecore.integration;

import magicbook.gtlitecore.api.module.Module;
import magicbook.gtlitecore.api.utils.GTLiteValues;
import magicbook.gtlitecore.core.module.BaseModule;
import magicbook.gtlitecore.core.module.GTLiteModules;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.jetbrains.annotations.NotNull;

import java.util.Collections;
import java.util.List;

@Module(moduleId = GTLiteModules.MODULE_INTEGRATION,
        containerId = GTLiteValues.MODID,
        name = "GregTech Lite Integration Module",
        descriptions = "General GregTech Lite Integration Module. Disabling this disables all integration modules.")
public class IntegrationModule extends BaseModule
{

    public static final Logger logger = LogManager.getLogger(GTLiteValues.NAME + " Mod Integration");

    @NotNull
    @Override
    public List<Class<?>> getEventBusSubscribers()
    {
        return Collections.singletonList(IntegrationModule.class);
    }

    @NotNull
    @Override
    public Logger getLogger()
    {
        return logger;
    }

}

File: src/main/kotlin/gregtechlite/gtlitecore/integration/IntegrationModule.kt
package gregtechlite.gtlitecore.integration

import gregtechlite.magicbook.util.SidedLogger
import gregtechlite.gtlitecore.api.MOD_ID
import gregtechlite.gtlitecore.api.module.Module
import gregtechlite.gtlitecore.core.module.BaseModule
import gregtechlite.gtlitecore.core.module.GTLiteModules.Companion.MODULE_INTEGRATION
import org.apache.logging.log4j.Logger

@Module(moduleId = MODULE_INTEGRATION,
        containerId = MOD_ID,
        name = "GregTech Lite Integration Module",
        descriptions = "General GregTech Lite Integration Module. Disabling this disables all integration modules.")
class IntegrationModule : BaseModule()
{

    companion object
    {

        @JvmField
        val logger: Logger = SidedLogger("$MOD_ID-integration")

    }

    override fun getEventBusSubscribers(): MutableList<Class<*>?>
    {
        return mutableListOf(IntegrationModule::class.java)
    }

    override fun getLogger(): Logger = Companion.logger

}
